rina-tools/src/rina-echo-time/et-client.cc

```diff
@@ -475,25 +513,15 @@ int Client::readSDU(int portId, void * sdu, int maxBytes, unsigned int timeout)
 {
 	timespec begintp, endtp;
 	get_current_time(begintp);
-	int bytes_read;
-
-	while (true) {
-		try {
-			bytes_read = ipcManager->readSDU(portId, sdu, maxBytes);
-			if (bytes_read == -EAGAIN) {
-				get_current_time(endtp);
-				if ((unsigned int) time_difference_in_ms(begintp, endtp) > timeout) {
-					break;
-				}
-			} else {
-				break;
-			}
-		} catch (rina::Exception &e) {
-			throw e;
-		}
-	}
+	ReadSDUWithTimeout _(portId, timeout);
+	do {
+		int bytes_read = ipcManager->readSDU(portId, sdu, maxBytes);
+		if (bytes_read != -EINTR)
+			return bytes_read;
+		get_current_time(endtp);
+	} while ((unsigned) time_difference_in_ms(begintp, endtp) < timeout);
```

Comment:
The type of `timeout` is already `unsigned`, so I guess you mean `_difference_in_ms(begintp, endtp) < timeout`.
I don't know why there was cast to an integral type; my commit does not change the condition (well, off by 1ms). A more important issue is the use of `get_current_time` (i.e. `gettimeofday`), whereas we'd prefer to use a monotonic clock for timeouts.